routing/pathfind.go

```diff
@@ -228,13 +236,9 @@ type graphParams struct {
 // RestrictParams wraps the set of restrictions passed to findPath that the
 // found path must adhere to.
 type RestrictParams struct {
-	// IgnoredNodes is an optional set of nodes that should be ignored if
-	// encountered during path finding.
-	IgnoredNodes map[route.Vertex]struct{}
-
-	// IgnoredEdges is an optional set of edges that should be ignored if
-	// encountered during path finding.
-	IgnoredEdges map[EdgeLocator]struct{}
+	// ProbabilitySource is a callback that is expected to return the
+	// success probability of traversing the channel from the node.
+	ProbabilitySource func(route.Vertex, EdgeLocator) float64
```

Comment:
It is indeed a hard restriction anymore, but a cost factor. Although probabilities can still be 0, in which case it would be a restriction again. To me it is mainly semantics. I prefer to leave it here and if more cost factor controls are added, extract it then.